### ISSUE TITLE
build GRASS

### ISSUE THREAD

**Issue by alaniwi** [cedadev/jasmin_scivm#17]
https://trac.osgeo.org/gdal/wiki/GRASS

**Comment by alaniwi** [comment]
(per CEDA footprints ticket 29091)

**Comment by danclewley** [comment]
This ticket should be 'Build GRASS' not 'build GDAL with GRASS support' as per the CEDA ticket (however, a prerequisite of building GDAL with GRASS support is to build GRASS). Although GRASS is already available as a package for CentOS 6 from the standard repositories it conflicts with the CEDA build of GDAL.

**Comment by alaniwi** [comment]
now per footprints 31872

**Comment by alaniwi** [comment]
In general there are RedHat packages for grass, grass-devel, grass-libs, but they depend on geos-3.3.2 (exact version). We provide geos 3.3.6 as part of JAP, with several things depending on it (Shapely, cartopy, gdal). Separate ticket exists to upgrade gdal in any case (https://github.com/cedadev/jasmin_scivm/issues/6). Probably ought really to build latest versions of all of these and not rely on RedHat packages.

**Comment by alaniwi** [comment]
geos 3.5.0 built. As expected, not compatible with existing gdal. Shapely and cartopy don't mind because the major lib version of libgeos_c.so is still 1.

**Comment by alaniwi** [comment]
Would like to use RH SRPM (grass-6.4.4-6.el6.src.rpm) as starting point.
Trying to find it (ftp://ftp.redhat.com/redhat/linux/beta/6Server/en/os/SRPMS/ empty).
Fedora SRPM is at http://www.mirrorservice.org/sites/dl.fedoraproject.org/pub/fedora/linux/development/rawhide/source/SRPMS/g/grass-6.4.4-9.fc24.src.rpm

**Comment by alaniwi** [comment]
Trying to rebuild grass from RedHat SRPM, claims various build time dependencies, some of which are straightforwardly "yum install"-ed, leaving:

```
        gdal-devel is needed by grass-6.4.4-1.ceda.el6.x86_64
        proj-epsg is needed by grass-6.4.4-1.ceda.el6.x86_64
        proj-nad is needed by grass-6.4.4-1.ceda.el6.x86_64
```

Binary RedHat packages for proj-epsg and proj-nad will not work because they depend on proj 4.7.0 and we have a CEDA build of proj 4.8.0, so will need to build these from source.

Dependency on gdal-devel.  This is consistent with the order of steps shown at https://trac.osgeo.org/gdal/wiki/GRASS, based on which it seems best to build a gdal RPM without GRASS support (ticket https://github.com/cedadev/jasmin_scivm/issues/6), then build grass.

After that, the GRASS wiki suggests a choice of rebuilding GDAL with GRASS support or building the plugin, and it seems sensible to build the plugin as a separate RPM. This should give a nice linear dependency chain, and avoid the confusion of producing 2 RPMs both called "gdal", one of which lacks GRASS support and is only installed temporarily during build.

**Comment by alaniwi** [comment]
proj-epsg and proj-nad now installed on builder (already built from same SRPM as proj, just weren't installed...)

**Comment by alaniwi** [comment]
GDAL now built. RedHat SRPM can be built, but includes Python components, and we want python 2.7, so attempting to build GRASS for python 2.7 based on RedHat SRPM.  Going to require python2.7 build of wxPython as prerequisite.

**Comment by alaniwi** [comment]
wxPython needs gstreamer

**Comment by alaniwi** [comment]
Trying unsuccessfully to build rebuild wxPython-3.0.2.0-6.fc23.src.rpm (initially without changes, though with intent to change to python 2.7). Seems not to find relevant includes.

Trying instead more of a vanilla build from source tarball.

**Comment by alaniwi** [comment]
Got python27-wxPython built. Python 2.6 version will need uninstalled to allow the 2.7 version to have paths in /usr/bin without version suffixes (such as `/usr/bin/editra`) and without conflicts. However, nothing else is depending on wxPython so let's do this.

**Comment by alaniwi** [comment]
grass RPM failing to build. Complains of error in `/home/builderdev/rpmbuild/BUILD/grass-6.4.4/gui/wxpython`. Appears to be trying to use system python instead of 2.7. Offending command:
```
[builderdev@builder wxpython]$ make
GISRC=/home/builderdev/rpmbuild/BUILD/grass-6.4.4/dist.x86_64-redhat-linux-gnu/demolocation/.grassrc64 GISBASE=/home/builderdev/rpmbuild/BUILD/grass-6.4.4/dist.x86_64-redhat-linux-gnu PATH="/home/builderdev/rpmbuild/BUILD/grass-6.4.4/dist.x86_64-redhat-linux-gnu/bin:$PATH" PYTHONPATH="/home/builderdev/rpmbuild/BUILD/grass-6.4.4/dist.x86_64-redhat-linux-gnu/etc/python:$PYTHONPATH" LD_LIBRARY_PATH="/home/builderdev/rpmbuild/BUILD/grass-6.4.4/dist.x86_64-redhat-linux-gnu/bin:/home/builderdev/rpmbuild/BUILD/grass-6.4.4/dist.x86_64-redhat-linux-gnu/lib::" LC_ALL=C python core/menudata.py > menustrings.py
Traceback (most recent call last):
  File "core/menudata.py", line 39, in <module>
    import wx
ImportError: No module named wx
make: *** [menustrings.py] Error 1
```

**Comment by alaniwi** [comment]
Looks like I simply forgot to tell it to use 2.7 (aside from in the dependencies). Seems the top-level configure accepts `--with-python=/usr/bin/python2.7`, so will try that. If that fails, appears that `make PYTHON=python2.7` works in the `gui/wxpython` directory, though not tried passing it via the top-level make.

**Comment by alaniwi** [comment]
With full path, falls over early in configure with not finding `Python.h` even though installed python2.7 RPM includes devel stuff, but it gets past that point if using just `--with-python=python2.7`

**Comment by alaniwi** [comment]
grass RPM now builds cleanly (though forgot to change a runtime dependency to 2.7 version, so need to rebuild).

**Comment by alaniwi** [comment]
RPM built and installing and running "grass" executable launches the GUI although necessary to set environment variable `GRASS_PYTHON=python2.7`. Will just document this fact for users, as it's not clear how to remove the need, having already told the build process to use that python version.

Ready for installation with next updates, though may bring forward installation on jasmin-sci1-test given that user waiting.
